### PR TITLE
EZP-26209 Textline legacy search indexing causes TransactionError

### PR DESCRIPTION
**Pull request by wawrzynkiewicz**
Causes SQL error: #1264 - Out of range value for column 'integer_value' at row 1
@integerValue generation the value needs to be verified against MySQL's max int value.

### CODE REVIEW

**Comment by yannickroger** [comment]
ping @joaoinacio 
Also keep in mind that we also support PGsql.

**Comment by blankse** [comment]
+1 I have the same error

**Comment by pkamps** [comment]
I run into the same issue.

File: kernel/search/plugins/ezsearchengine/ezsearchengine.php [hunk 1]
```diff
@@ -96,7 +96,11 @@ public function addObject( $contentObject, $commit = true )
                     // Split text on whitespace
                     if ( is_numeric( trim( $text ) ) )
                     {
-                        $integerValue = (int) $text;
+                        //verify if value does not exceed MySQL int(11) unsigned maximum value
```
Comment (adamwojs) [inline kernel/search/plugins/ezsearchengine/ezsearchengine.php]: Based on magic number one line below, I think that you have on your mind the maximum **signed** value of `int`. MySQL documentation says (https://dev.mysql.com/doc/refman/5.7/en/integer-types.html) that maximum unsigned value of int is `4294967295` 😉

**Comment by peterkeung** [comment]
What is the "integer_value" field used for anyway? That would determine whether setting it to "0" (rather than the max value, for example) is an appropriate action.

**Comment by blankse** [comment]
It is possible to define a searchtype.
https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L1356

In the types integer, integers, byrange, byidentifierrange, integersbyidentifier the integer_value is used.

I would prefer the changes from @pkamps in his fork :)

**Comment by andrerom** [comment]
@blankse Which changes do you refer to?

**Comment by blankse** [comment]
@andrerom Yes, the linked one.

It set the max value and takes the min value into account.

**Comment by andrerom** [comment]
@blankse with only nitpic of not referring just to mysql on the constants, I agree, it's cleaner approach. Maybe something that can be opened here?

**Comment by andrerom** [comment]
Closing, handled in #1338

**Pull request closed without merging**